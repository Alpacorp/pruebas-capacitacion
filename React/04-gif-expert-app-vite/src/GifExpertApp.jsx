import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["God of War"]);

  const onAddCategory = (newCategory) => {
    const lowerCaseCategory = newCategory.toLowerCase();
    const existentCategory = categories.map((category) =>
      category.toLowerCase()
    );

    if (existentCategory.includes(lowerCaseCategory)) return;
    setCategories((category) => [newCategory, ...category]);
  };

  return (
    <div>
      <h1>GifExpertApp</h1>
      <AddCategory
        onNewCategory={(newCategory) => onAddCategory(newCategory)}
      />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </div>
  );
};

export default GifExpertApp;
