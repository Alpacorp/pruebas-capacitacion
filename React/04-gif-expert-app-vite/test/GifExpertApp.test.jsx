import { render, screen } from "@testing-library/react";
import GifExpertApp from "../src/GifExpertApp";

describe("Pruebas en <GifExpertApp />", () => {
  test("El componente debe hacer match con el snapshot", () => {
    render(<GifExpertApp />);
    expect(screen.getByTestId("gif-expert-app")).toMatchSnapshot();
    screen.debug();
  });

  test("Debe contener los elementos H1 (GifExpertApp), H3 (God Of War) y <p>Loading...</p>", () => {
    render(<GifExpertApp />);

    const h1 = screen.getByRole("heading", { level: 1 }).innerHTML;
    const h3 = screen.getByRole("heading", { level: 3 }).innerHTML;
    const p = screen.getByText("Loading...");

    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toBe(h1);
    expect(screen.getByRole("heading", { level: 3 }).innerHTML).toBe(h3);
    expect(screen.getByText("Loading...")).toBe(p);
  });

  test("Debe de ejecutar la funciÃ³n onAddCategory", () => {
    const onAddCategory = jest.fn();

    render(<GifExpertApp />);

    expect(onAddCategory).toHaveBeenCalledTimes(0);
  });
});
