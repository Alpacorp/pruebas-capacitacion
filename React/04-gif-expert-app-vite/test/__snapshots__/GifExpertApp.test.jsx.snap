// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> El componente debe hacer match con el snapshot 1`] = `
<div
  data-testid="gif-expert-app"
>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    God of War
  </h3>
  <p>
    Loading...
  </p>
  <div
    class="card-grid"
  />
</div>
`;
